enhancements/agent-installer/host-based-mirror.md

```diff
@@ -0,0 +1,282 @@
+---
+title: host-based-mirror
+authors:
+  - "@bfournie"
+reviewers:
+  - "@andfasano"
+  - "@patrickdillon"
+  - "@romfreiman"
+  - "@rwsu"
+approvers:
+  - "@zaneb"
+api-approvers:
+  - None
+creation-date: 2023-02-03
+last-updated: 2023-02-16
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AGENT-262
+---
+
+# Host Based Mirror
+
+## Summary
+
+The agent based installer should have the ability to include all container
+images needed to boot a host in a disconnected environment, along with
+necessary tools, when it generates an ISO. This will allow the hosts to boot
+without an external mirror registry residing on a separate server.
+
+## Motivation
+
+In a disconnected environment, a user must set up and maintain a local registry
+on a separate server in order to provide container images to the hosts being
+booted. In addition to the extra hardware, this step requires that the user be
+familiar with both how to create the registry and the tools used to mirror the
+images, e.g. [oc-mirror](https://github.com/openshift/oc-mirror).  The content
+of the local registry must also be synchronized with the releaseImage provided
+in the manifests files; whenever there is an update to the releaseImage the
+local registry must also be updated.  Finally the user must add an
+ImageContentSources section to the install-config.yaml to define the registry
+sources and mirrors. This is error-prone and often results in misconfiguration
+and failed installations.
+
+This enhancement would remove the need for an additional server and
+reduce the burden on the user when using the agent based installer in a
+disconnected environment.
+
+### User Stories
+
+- As a user creating an initial Openshift cluster in a disconnected
+  environment I do not want to create and manage a registry to provide
+  images for the cluster. As I may not have access to an an external server
+  to run a registry, I need to boot the hosts for the initial cluster with no
+  additional servers.
+- As a user with limited Openshift experience I want to use an ISO that
+  fully contains all that is needed to create a cluster.
+
+### Goals
+
+- Remove the need to set up and maintain a local registry in fully-disconnected
+  environments.
+- Include in the ISO the container images and tools required for mirroring.
+- Provide an additional command to build a container image tarfile using
+  the embedded releaseImage in a connected environment.
+- When building the ISO in a disconnected environment, allow for a tarfile,
```

Comment:
The thought is that we provide both for the generation of the image tarfile, in which case the create image command must be run in a connected environment, and for an image tarfile provided by the user (i.e. created separately using oc-mirror) so that the create image command could be run in a disconnected environment.